token/services/ttxdb/db/badger/badger.go

```diff
@@ -512,6 +521,91 @@ func (db *Persistence) GetStatus(txID string) (driver.TxStatus, error) {
 	return driver.Unknown, errors.Errorf("transaction [%s] not found", txID)
 }
 
+func (db *Persistence) AddTransactionEndorsementAck(txID string, id view.Identity, sigma []byte) error {
+	key, err := db.transactionEndorseAckKey(txID, id)
+	if err != nil {
+		return errors.Wrapf(err, "could not get key for token request %s", txID)
+	}
+	logger.Debugf("adding transaction endorsement ack [%s] with key [%s]", txID, key)
+
+	value := &TransactionEndorseAck{
+		TxID:  txID,
+		ID:    id,
+		Sigma: sigma,
+	}
+
+	b, err := MarshalTransactionEndorseAck(value)
+	if err != nil {
+		return errors.Wrapf(err, "could not marshal record for key %s", key)
+	}
+
+	txn := db.db.NewTransaction(true)
+	defer txn.Discard()
+	err = txn.Set([]byte(key), b)
+	if err != nil {
+		return errors.Wrapf(err, "could not set value for key %s", key)
+	}
+	err = txn.Commit()
+	if err != nil {
+		return errors.Wrapf(err, "could not commit transaction for [%s]", txID)
+	}
+
+	return nil
+}
+
+func (db *Persistence) GetTransactionEndorsementAcks(txID string) (map[string][]byte, error) {
+	acks := make(map[string][]byte)
+
+	// search for all matching keys
+	type Entry struct {
+		key   string
+		value []byte
+	}
+	var entries []Entry
+	stream := db.db.NewStream()
+	stream.NumGo = db.numGoStream
+	stream.LogPrefix = streamLogPrefixStatus
+	txIdAsBytes := []byte(txID)
+	stream.ChooseKey = func(item *badger.Item) bool {
```

Comment:
If I understand correctly, you only choose the "tea"-prefixed entries later. In this case I see that badger has a `Prefix` property that I think filters out more efficiently. This way you skip the check later and you iterate over less keys.